client/FxAccountClient.js

```diff
@@ -568,27 +572,51 @@ define([
   FxAccountClient.prototype.accountReset = function(email, newPassword, accountResetToken, options) {
     var self = this;
     var data = {};
+    var unwrapBKey;
 
     options = options || {};
 
     if (options.metricsContext) {
       data.metricsContext = metricsContext.marshall(options.metricsContext);
     }
 
+    if (options.sessionToken) {
+      data.sessionToken = options.sessionToken;
+    }
+
     required(email, 'email');
     required(newPassword, 'new password');
     required(accountResetToken, 'accountResetToken');
 
     return credentials.setup(email, newPassword)
       .then(
         function (result) {
+          if (options.keys) {
+            unwrapBKey = sjcl.codec.hex.fromBits(result.unwrapBKey);
+          }
+
           data.authPW = sjcl.codec.hex.fromBits(result.authPW);
 
           return hawkCredentials(accountResetToken, 'accountResetToken',  HKDF_SIZE);
         }
       ).then(
         function (creds) {
-          return self.request.send('/account/reset', 'POST', creds, data);
+          var queryParams = '';
+          if (options.keys) {
+            queryParams = '?keys=true';
```

Comment:
Just added [this check up here](https://github.com/mozilla/fxa-js-client/pull/197/files#diff-9f20221066a888d5c711f89f8e165858R524).